Zapper-fi/studio
# feat(ferro): add ferro app integration with vvs finance refactoring

## Description

### 1. refactor [VVS finance](https://vvs.finance/) to share xVault fetcher

https://github.com/Zapper-fi/studio/commit/5713c88122f3602cf99adfb199b4dca451907c29

* refactor into helpers: `fetchXVaultPositions`, `getXVaultBalances`
* correct `VVS_FINANCE_DEFINITION.groups.xvvsVault.type` to `GroupType.POSITION`

### 2. add [Ferro](https://ferroprotocol.com/) integration

* pool/lp token fetcher (currently only 3FER)
* liquidity staking position fetcher
* xFER token fetcher
* xFER vaults fetcher
  * using refactored helper from VVS finance
* balance fetcher

## Checklist

- [x] I have followed the [Contributing Guidelines](https://github.com/Zapper-fi/studio/blob/edbdd22/CONTRIBUTING.md)
- [x] (optional) As a contributor, my Twitter handle is: https://twitter.com/cronos_labs

## How to test?

1. edit/create `.env` adding `ENABLED_APPS="ferro"`
2. run `pnpm dev` to spin up nest server
3. visit dev endpoints below to see result of fetchers to integrate ferro:
  * http://localhost:5001/apps/ferro/tokens?groupIds[]=pool&network=cronos
  * http://localhost:5001/apps/ferro/positions?groupIds[]=staked-liquidity&network=cronos
  * http://localhost:5001/apps/ferro/tokens?groupIds[]=xfer&network=cronos
  * http://localhost:5001/apps/ferro/positions?groupIds[]=xfer-vault&network=cronos
  * `http://localhost:5001/apps/ferro/balances?addresses[]=<wallet_address>&network=cronos`
  * http://localhost:5001/apps/ferro/tvl?network=cronos
